sdks/python/apache_beam/examples/snippets/transforms/element_wise/pardo_test.py

```diff
@@ -58,24 +58,44 @@ def check_dofn_params(actual):
 type(window) -> <class 'apache_beam.transforms.window.IntervalWindow'>
 window.start -> Timestamp(1584675660) (2020-03-20 03:41:00)
 window.end -> Timestamp(1584675690) (2020-03-20 03:41:30)
-window.max_timestamp() -> Timestamp(1584675689.999999) (2020-03-20 03:41:29.999999)'''
-  # [END dofn_params]
+window.max_timestamp() -> Timestamp(1584675689.999999) (2020-03-20 03:41:29.999999)
+[END dofn_params]'''.splitlines()[1:-1])
   # pylint: enable=line-too-long
   assert_that(actual, equal_to([dofn_params]))
 
 
+def check_dofn_methods(actual):
+  results = '''[START results]
+__init__
+setup
+start_bundle
+* process: 🍓
+* process: 🥕
+* process: 🍆
+* process: 🍅
+* process: 🥔
+* finish_bundle: 🌱🌳🌍
+teardown
+[END results]'''.splitlines()[1:-1]
+  results = [line for line in results if line.startswith('*')]
+  assert_that(actual, equal_to(results))
```

Comment:
Does this only validate that, process lines are in the output? Does it check for other lines?

Reply:
Good point. The sample code prints that as an output, and it makes sense to see what gets called after what. However, for the test we're only testing for the elements produced by `process` and `finish_bundle` since those are the ones that affect the output `PCollection`. So that's why I marked them with a `*` to make them easier to spot and test for. I'm adding a comment for this.

Reply:
Ping on this comment?

Reply:
I'm changing the test to actually check for both the order of the methods executed, as well as all the elements being present without ordering constraints.

Regarding if `DoFn` methods work on all runners, looking at #7994 I could only find that [BEAM-7885 `DoFn.setup` doesn't run for streaming jobs in the DirectRunner](https://issues.apache.org/jira/browse/BEAM-7885), and [BEAM-7340 `DoFn.teardown` metrics are lost](https://issues.apache.org/jira/browse/BEAM-7340). I'm adding those into comments.

I looked into Jira too, but that was all that I could find, are you aware of any other issues with other runners?